### PR TITLE
Add asset position

### PR DESCRIPTION
**Pull request by acbeni**
support for our backend feature https://github.com/sphereio/dev.commercetools.com/pull/740

### CODE REVIEW

**Comment by acbeni** [comment]
Closes #1613

File: commercetools-models/src/main/java/io/sphere/sdk/categories/Category.java [hunk 1]
```diff
@@ -185,6 +186,15 @@
 
     @Nonnull
     @IgnoreInQueryModel
+    @HasUpdateAction(value = "setAssetKey", fields = {@PropertySpec(name = "assetId", fieldType = String.class), @PropertySpec(name = "assetKey", fieldType = String.class, isOptional = true)})
```
Comment (katmatt) [inline commercetools-models/src/main/java/io/sphere/sdk/categories/Category.java]: Now we finally have an example for this usage of our annotations 👍

File: commercetools-models/src/main/java/io/sphere/sdk/categories/commands/updateactions/AddAsset.java [hunk 2]
```diff
@@ -1,25 +0,0 @@
-package io.sphere.sdk.categories.commands.updateactions;
```
Comment (katmatt) [inline commercetools-models/src/main/java/io/sphere/sdk/categories/commands/updateactions/AddAsset.java]: Nice that you removed more manually created update actions 👍

File: commercetools-models/src/main/java/io/sphere/sdk/categories/commands/updateactions/SetAssetCustomType.java [hunk 3]
```diff
@@ -17,21 +17,35 @@
...
+
     public static SetAssetCustomType of(final String assetId, @Nullable final CustomFieldsDraft customFieldsDraft) {
+        return of(assetId,null, customFieldsDraft);
+    }
+
+    public static SetAssetCustomType ofKey(final String assetKey, @Nullable final CustomFieldsDraft customFieldsDraft) {
+        return of(null,assetKey, customFieldsDraft);
+    }
+
+    private static SetAssetCustomType of(final String assetId,final String key, @Nullable final CustomFieldsDraft customFieldsDraft) {
```
Comment (katmatt) [inline commercetools-models/src/main/java/io/sphere/sdk/categories/commands/updateactions/SetAssetCustomType.java]: For consistency, I would suggest to rename the `key` parameter to `assetKey` here.

File: commercetools-models/src/main/java/io/sphere/sdk/models/AssetDraft.java [hunk 4]
```diff
@@ -21,7 +21,11 @@
         abstractBuilderClass = true,
         copyFactoryMethods = @CopyFactoryMethod(Asset.class),
         factoryMethods = @FactoryMethod(parameterNames = { "sources", "name"}))
-public interface AssetDraft extends CustomDraft {
+public interface AssetDraft extends CustomDraft, WithKey{
```
Comment (katmatt) [inline commercetools-models/src/main/java/io/sphere/sdk/models/AssetDraft.java]: Nice that you used our `WithKey` interface here 👍

File: commercetools-sdk-base/src/main/java/io/sphere/sdk/annotations/HasUpdateAction.java [hunk 5]
```diff
@@ -37,5 +37,10 @@
     Class[] superInterfaces() default {};
     boolean makeAbstract() default false;
 
+    /**
+     * Specifies whether there should be a default factory method generated.
```
Comment (katmatt) [inline commercetools-sdk-base/src/main/java/io/sphere/sdk/annotations/HasUpdateAction.java]: Would be nice if we could add a short description of what a default factory method is. Is it a factory method which provides all update action properties as parameters?

File: commercetools-models/src/test/java/io/sphere/sdk/products/commands/ProductUpdateCommandIntegrationTest.java [hunk 6]
```diff
@@ -1153,6 +1154,9 @@ public void productProjectionCanBeUsedToUpdateAProduct() throws Exception {
 
             final ProductProjection productProjection = client().executeBlocking(ProductProjectionByIdGet.of(product, STAGED));
 
+            SphereJsonUtils.toJsonString(productProjection);
+            SphereJsonUtils.toJsonNode(productProjection).forEach(System.out::println);
+
```
Comment (katmatt) [inline commercetools-models/src/test/java/io/sphere/sdk/products/commands/ProductUpdateCommandIntegrationTest.java]: Do we need the two lines above? Or are they left overs from testing?

File: commercetools-models/src/main/java/io/sphere/sdk/categories/commands/updateactions/SetAssetCustomType.java (hunk 3 above)
Comment (acbeni) [reply commercetools-models/src/main/java/io/sphere/sdk/categories/commands/updateactions/SetAssetCustomType.java]: good catch 👍

File: commercetools-models/src/test/java/io/sphere/sdk/products/commands/ProductUpdateCommandIntegrationTest.java (hunk 6 above)
Comment (acbeni) [reply commercetools-models/src/test/java/io/sphere/sdk/products/commands/ProductUpdateCommandIntegrationTest.java]: yep 😢 
i forgot to remove them.

File: commercetools-sdk-base/src/main/java/io/sphere/sdk/annotations/HasUpdateAction.java (hunk 5 above)
Comment (acbeni) [reply commercetools-sdk-base/src/main/java/io/sphere/sdk/annotations/HasUpdateAction.java]: actually it provides all required update actions as parameter, this was done to have more control over the generation process.
But i agree i should add more javadoc there 👍

**Pull request merged**